lib/plans/plan_details.dart

```diff
@@ -142,15 +156,34 @@ class PlanCard extends StatelessWidget {
           isPro: isPro,
         ),
       );
-    } else {
-      // * Proceed to our own Checkout
-      await context.pushRoute(
-        Checkout(
-          plan: plan,
-          isPro: isPro,
-        ),
-      );
+    } else if (isDesktop()) {
+      final paymentMethods = await sessionModel.paymentMethodsv4();
+      final providers = paymentProvidersFromMethods(paymentMethods);
+      // if only one payment provider is returned, bypass the last checkout screen
+      if (providers.length == 1) {
```

Comment:
@atavism Can we make sure that if the provider is only one and it not stripe then it should still goes to next screen


 ```
 if (providers.length == 1 && providers[0].name.toPaymentEnum() == Providers.stripe) {
        final provider = providers[0];
        final providerName = provider.name.toPaymentEnum();
        // Note: as of now, Stripe is the only payment provider that collects email

        final redirectUrl = await sessionModel.paymentRedirectForDesktop(
            context,
            plan.id,
            "",
            providerName,
          );
          await openDesktopWebview(
              context: context,
              provider: providerName,
              redirectUrl: redirectUrl);
          return;
      }
```

Reply:
> Can we make sure that if the provider is only one and it not stripe then it should still goes to next screen

@jigar-f Yes, that's how it currently works, we are just checking if the payment provider is Stripe further down:

```kotlin
      if (providers.length == 1) {
        final provider = providers[0];
        final providerName = provider.name.toPaymentEnum();
        // Note: as of now, Stripe is the only payment provider that collects email
        if (providerName == Providers.stripe) {
          final redirectUrl = await sessionModel.paymentRedirectForDesktop(
            context,
            plan.id,
            "",
            providerName,
          );
          await openDesktopWebview(
              context: context, provider: providerName, redirectUrl: redirectUrl);
          return;
        }
      }
```

I made those changes and added a comment to make it more explicit Stripe is the only payment provider this applies to